Interop/Tensorflow/src/main/java/org/tribuo/interop/tensorflow/TensorFlowTrainer.java

```diff
@@ -0,0 +1,597 @@
+         */
+        TRIBUO_NATIVE,
+        /**
+         * Saves the model state inside a TensorFlow checkpoint, emits a {@link TensorFlowCheckpointModel}.
+         */
+        CHECKPOINT;
+    }
+
+    @Config(mandatory=true,description="Path to the protobuf containing the graph.")
+    private Path graphPath;
+
+    private GraphDef graphDef;
+
+    @Config(description="Test time batch size.")
+    private int testBatchSize = 16;
+
+    @Config(mandatory=true,description="Name of the output operation before the loss.")
+    private String outputName;
+
+    @Config(description="Name of the init operation.")
+    private String initName = Init.DEFAULT_NAME;
+
+    @Config(mandatory=true,description="Feature extractor.")
+    private FeatureConverter<T> featureConverter;
+
+    @Config(mandatory=true,description="Response extractor.")
+    private OutputConverter<T> outputConverter;
+
+    @Config(description="Minibatch size.")
+    private int minibatchSize = 1;
+
+    @Config(description="Number of SGD epochs to run.")
+    private int epochs = 5;
+
+    @Config(description="Logging interval to print out the loss.")
+    private int loggingInterval = 100;
+
+    @Config(mandatory=true,description="The gradient optimiser to use.")
+    private GradientOptimiser optimiserEnum;
+
+    @Config(mandatory=true,description="The gradient optimiser parameters.")
+    private Map<String,Float> gradientParams;
+
+    @Config(description="Saved model format.")
+    private TFModelFormat modelFormat = TFModelFormat.TRIBUO_NATIVE;
+
+    @Config(description="Checkpoint output directory.")
+    private Path checkpointPath;
+
+    @Config(description="Inter operation thread pool size. -1 uses the default TF value.")
+    private int interOpParallelism = 1;
+
+    @Config(description="Intra operation thread pool size. -1 uses the default TF value.")
+    private int intraOpParallelism = 1;
+
+    private int trainInvocationCounter = 0;
+
+    /**
+     * for olcut.
+     */
+    private TensorFlowTrainer() {}
+
+    /**
+     * Constructs a Trainer for a TensorFlow graph. Stores the model parameters inside the Tribuo model.
+     * @param graphPath Path to the graph definition on disk. Must have the necessary targets and placeholders.
+     * @param featureConverter The example converter to convert a Tribuo {@link Example} into a {@link Tensor}.
+     * @param outputConverter The output converter to convert a Tribuo {@link Output} into a {@link Tensor} and back. This encodes the output type.
+     * @param minibatchSize The minibatch size to use in training.
+     * @param epochs The number of SGD epochs to run.
+     * @param testBatchSize The minibatch size to use at test time.
+     * @param loggingInterval The logging interval. Set to -1 to quiesce the loss level logging.
+     * @throws IOException If the graph could not be loaded from the supplied path.
+     */
+    public TensorFlowTrainer(Path graphPath,
```

Comment:
These constructors seem ripe for a Builder.

Reply:
I completely agree, but I don't want to do it now when we've not figured out what our overall builder story is for Tribuo (e.g. OLCUT generated ones).